compiler/rustc_middle/src/traits/mod.rs

```diff
@@ -256,20 +249,20 @@ pub enum ObligationCauseCode<'tcx> {
 
     /// Must satisfy all of the where-clause predicates of the
     /// given item.
-    ItemObligation(DefId),
+    MiscItem(DefId),
```

Comment:
Alternative could be `WhereClause`, and then `SpannedWhereClause`, etc.

Reply:
Actually, I actually like `WhereClause` better

Reply:
yeah, I would like `Where` and `SpannedWhere`, or even better, use `Where(DefId, Option<Span>)`